src/main/scala/cromwell/engine/package.scala

```diff
@@ -43,19 +43,59 @@ package object engine {
     def randomId() = WorkflowId(UUID.randomUUID())
   }
 
+  object WorkflowDescriptor {
+    private def disabledMessage(readWrite: String, consequence: String) =
+      s"""$readWrite is enabled in the workflow options but Call Caching is disabled in this Cromwell instance.
+         |As a result the calls in this workflow $consequence
+       """.stripMargin
+
+    val writeDisabled = disabledMessage("Write to Cache", "WILL NOT be cached")
+    val readDisabled = disabledMessage("Read from Cache", "WILL ALL be executed")
```

Comment:
Fair enough.

I think it'd be better if down below you could say something like `if (optionCacheWriting) logWriteDisabled` where `logWriteDisabled` is a function (in the class due to needing the logger) which does the `workflowLogger.warn(writeDisabled)` bit. 

At the end of the day the only important information in that if clause is that it's going to log a message about the write being disabled and most people will likely never care what's actually being logged, so this would make that information more concise.